File: packages/contracts/src/health.ts
import { z } from 'zod';

export const HealthLiveResponseSchema = z.strictObject({
  status: z.literal('ok'),
});

export type HealthLiveResponse = z.infer<typeof HealthLiveResponseSchema>;

export const HealthReadyResponseSchema = z.strictObject({
  status: z.literal('ok'),
});

export type HealthReadyResponse = z.infer<typeof HealthReadyResponseSchema>;

File: apps/flutter/lib/core/contracts/health.dart
import 'json_reader.dart';

/// `HealthLiveResponseSchema` / `HealthReadyResponseSchema`: both are
/// `{ status: 'ok' }` strict objects.
class HealthResponse {
  const HealthResponse._();

  static HealthResponse fromJson(Object? json) {
    final reader = JsonReader.object(json);
    final status = reader.string('status');
    if (status != 'ok') {
      throw ContractViolationException('status', "expected literal 'ok'");
    }
    return const HealthResponse._();
  }
}
